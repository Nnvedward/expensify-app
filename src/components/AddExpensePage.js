import React from "react"
import { connect } from "react-redux"
import ExpenseForm from "./ExpenseForm"
import { addExpense } from "../actions/expenses"
import { useNavigate } from "react-router-dom"

const AddExpensePage = (props) => {
    const history = useNavigate()
    return (
        <div>
            <h1>Add Expense</h1>
            <ExpenseForm
                handleSubmit={(expense) => {
                    props.dispatch(addExpense(expense))
                    history('/')
                }} />
        </div>
    )
}

export default connect()(AddExpensePage)