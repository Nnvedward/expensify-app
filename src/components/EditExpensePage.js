import React from "react";
import { useParams } from 'react-router-dom'

const EditExpensePage = () => {
    const { id } = useParams()
    return (
        <div>
            Editing the expenses with the id of {id}
        </div>
    )
}

export default EditExpensePage