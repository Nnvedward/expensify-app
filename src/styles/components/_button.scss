.button {
    background: $dark-grey;
    color: white;
    padding: $s-size;
    border: none;
    font-weight: 300;
    font-size: $font-size-large;
    display: inline-block;
    text-decoration: none;
    line-height: 1;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #ad0a0a;
}