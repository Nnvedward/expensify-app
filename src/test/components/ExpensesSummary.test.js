import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test('Should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={234}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={32} expensesTotal={2345887189}/>)
    expect(wrapper).toMatchSnapshot()
})