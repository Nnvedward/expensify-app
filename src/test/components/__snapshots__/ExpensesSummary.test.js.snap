// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.34
  </h1>
</div>
`;

exports[`Should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    32
     
    expenses
     totalling 
    $23,458,871.89
  </h1>
</div>
`;
